drivers/Ticker.h

```diff
@@ -74,6 +79,30 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
     // When low power ticker is in use, then do not disable deep sleep.
     Ticker(const ticker_data_t *data);
 
+    /** Attach a function to be called by the Ticker, specifying the interval in std::chrono::duration
+     *
+     *  @param func pointer to the function to be called
+     *  @param t the time interval between calls
+     *
+     *  @note setting @a t to a value shorter than it takes to process the ticker callback
+     *  causes the system to hang. Ticker callback is called constantly with no time
+     *  for threads scheduling.
+     *
+     */
+#if __cplusplus >= 201103 && !defined __CC_ARM
+    template <typename Rep, typename Period>
+    void attach(Callback<void()> func, std::chrono::duration<Rep, Period> t)
```

Comment:
I also had second thoughts about using templates and very happy to use 
```
void attach(Callback<void()> func, std::chrono::microseconds t);
```

There will be no time to work on `ThisThread::sleep_for` in this release. Also because there are other time related functions like `ThisThread::sleep_until`